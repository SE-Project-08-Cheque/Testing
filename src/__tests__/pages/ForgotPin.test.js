// import dependencies
import React from "react";
// import react dom
import ReactDOM from "react-dom";
// import react-testing methods
import { render, cleanup, fireEvent } from "@testing-library/react";
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import ForgotPin from "../../pages/FogotPin";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ForgotPin />, div);
});

it("renders forgot pin page correctly", () => {
  const { queryByTestId } = render(<ForgotPin />);
    expect(queryByTestId("forgotpin-text")).toBeInTheDocument();
});

it("check button", () => {
    const { queryByTestId } = render(<ForgotPin />);
    fireEvent.click(queryByTestId("getpin-button"));
})


it("matches snapshot", () => {
  const tree = renderer.create(<ForgotPin />).toJSON();
  expect(tree).toMatchSnapshot();
});
