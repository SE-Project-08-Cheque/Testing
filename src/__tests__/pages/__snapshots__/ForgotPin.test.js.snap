// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
>
  <div
    className="css-ure0nq"
  >
    <div
      className="css-dbpzvv"
    >
      <div
        className="css-1rul54"
      >
        <div
          className="css-gmuwbf"
        >
          <p
            className="chakra-text css-kbn7if"
          >
            
          </p>
        </div>
        <div
          className="css-sc5c3h"
        >
          <span
            className="chakra-badge css-ifsvay"
          >
            Online
          </span>
        </div>
        <div
          className="css-16n1mrt"
        >
          <div
            aria-valuemax={100}
            aria-valuemin={0}
            className="css-12om7f"
            data-indeterminate=""
            role="progressbar"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
      </div>
      <div
        className="css-8atqhb"
      >
        <div
          className="chakra-stack css-zsoya5"
        >
          <a
            href="/home"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="overview"
              type="button"
            >
              Overview
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/Account_Details"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="personal_details"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Personal Details
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/Contact_Details"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="contact_details"
              type="button"
            >
              Contact Details
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/AllNotifications"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="notification"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Notification
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/ForgotPin"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="forgot_pin"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Fogot Pin
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <hr
            aria-orientation="horizontal"
            className="chakra-divider css-1upb9tn"
            role="separator"
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    <main
      className="MuiContainer-root MuiContainer-maxWidthXs"
    >
      <div
        className="makeStyles-paper-1"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <h1
          className="MuiTypography-root MuiTypography-h5"
          data-testid="forgotpin-text"
        >
          Fogot Pin
        </h1>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p>
            After Click Fogot Pin You will receive notification on new pin. Check Notification Bar
          </p>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
          data-testid="getpin-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Get New Pin
          </span>
        </button>
      </div>
      <div
        className="MuiBox-root MuiBox-root-8"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        >
          Copyright Â© 
          2021, All Rights Reserved | Designed by University of Moratuwa
        </p>
      </div>
    </main>
  </div>
</div>
`;
